FMSgridtools/make_hgrid/make_hgrid.py

```diff
@@ -1 +1,1250 @@
+          "be specified when --nest_grid(s) is set."),
+)
+@click.option(
+    "--use_great_circle_algorithm", 
+    is_flag=True, 
+    default=False,
+    help="When specified, great_circle_algorithm will be used to compute grid \
+         cell area.",
+)
+@click.option(
+    "--out_halo", 
+    type=int, 
+    default=0,
+    help="extra halo size data to be written out. This is only works for \
+         gnomonic_ed.",
+)
+@click.option(
+    "--no_output_length_angle",
+    is_flag=True, 
+    default=False,
+    help="When specified, will not output length(dx,dy) and angle \
+         (angle_dx, angle_dy)"
+)
+@click.option(
+    "--angular_midpoint", 
+    is_flag=True,
+    default=False,
+    help=""
+)
+@click.option(
+    "--rotate_poly",
+    is_flag=True, 
+    default=False,
+    help=("Set to calculate polar polygon areas by calculating the area of a "
+          "copy of the polygon, with the copy being rotated far away from the" 
+          "pole.")
+)
+@click.option(
+    "--verbose", 
+    is_flag=True, 
+    default=False,
+    help=("Will print out running time message when this option is set. "
+          "Otherwise the run will be silent when there is no error.")
+)
+def make_hgrid(
+    grid_type: str,
+    my_grid_file: Optional[str],
+    nxbnds: int,
+    nybnds: int,
+    xbnds: str,
+    ybnds: str,
+    nlon: str,
+    nlat: str,
+    dlon: str,
+    dlat: str,
+    lat_join: float,
+    nratio: int,
+    simple_dx: float,
+    simple_dy: float,
+    grid_name: str,
+    center: str,
+    shift_fac: float,
+    f_plane_latitude: float,
+    do_schmidt: bool,
+    do_cube_transform: bool,
+    stretch_factor: float,
+    target_lon: float,
+    target_lat: float,
+    nest_grids: int,
+    parent_tile: str,
+    refine_ratio: str,
+    istart_nest: str,
+    iend_nest: str,
+    jstart_nest: str,
+    jend_nest: str,
+    halo: int,
+    use_great_circle_algorithm: bool,
+    out_halo: int,
+    no_output_length_angle: bool,
+    angular_midpoint: bool,
+    rotate_poly: bool,
+    verbose: bool,
+):
+    nratio = 1
+    method = "conformal"
+    orientation = "center_pole"
+    nxbnds0 = nxbnds
+    nybnds0 = nybnds
+    nxbnds1 = 0
+    nybnds1 = 0
+    nxbnds2 = 0
+    nybnds2 = 0
+    nxbnds3 = 0
+    nybnds3 = 0
+
+    num_nest_args = 0
+    nn = 0
+
+    present_stretch_factor = 0
+    present_target_lon = 0
+    present_target_lat = 0
+    output_length_angle = 1
+    ntiles = 1
+    ntiles_global = 1
+    ntiles_file = 0
+    grid_obj = HGridObj()
+
+    center = "none"
+    geometry = "spherical"
+    projection = "none"
+    arcx = "small_circle"
+    north_pole_tile = "0.0 90.0"
+    north_pole_arcx = "0.0 90.0"
+    discretization = "logically_rectangular"
+    conformal = "true"
+
+    fms_file = "input.nml"
+    with open(fms_file, "x") as file:
+        file.write("")
+
+    #TODO: Will change after pyFMS refactor
+    pyfms = pyFMS(cFMS_path="./pyFMS/cFMS/libcFMS/.libs/libcFMS.so")
```

Comment:
this will change very soon.  let's wait until the library auto-loading capability is added in pyFMS before merging this PR in.